### PR TITLE
[1.1] Update buildtools and cli version to same ones as CoreFX

### PR DESCRIPTION
**Pull request by ianhays**
Having separate CLI version greatly increases the cost of adding new distros, so we should keep coreclr and corefx using the same CLI and buildtools version for their tools. 

Also, this CLI currently being used in coreclr 1.1 is *old*.

### CODE REVIEW

File: init-tools.sh [hunk 1]
```diff
@@ -82,7 +82,7 @@ if [ "$OS" == "OSX" ]; then
   ulimit -n 2048
 fi
 
-__CLIDownloadURL=https://dotnetcli.blob.core.windows.net/dotnet/preview/Binaries/${__DOTNET_TOOLS_VERSION}/${__DOTNET_PKG}.${__DOTNET_TOOLS_VERSION}.tar.gz
```
Comment (janvorli) [inline init-tools.sh]: I would rather copy the init-tools.sh/cmd  and Tools-Override folder from the corefx too so that we have complete match.

Comment (ianhays) [reply init-tools.sh]: They've slightly different in behavio. For example, coreclr tries to restore a CLI  based on the `uname` results directly, whereas corefx has a switch statement with fallbacks if it doesn't know about a particular RID.

If we're fine adopting that behavior in coreclr, then I see no problem bringing over the entire init-tools.* scripts from the corefx repo.

Comment (janvorli) [reply init-tools.sh]: Ah, I've missed that. Ok, let's stay on the safe side for now and keep what you have.

**Comment by janvorli** [comment]
Hmm, it looks like there is a problem:
```
17:26:49 curl --retry 10 -sSL --create-dirs -o /mnt/resource/j/workspace/dotnet_coreclr/release_1.1.0/debug_centos7.1_prtest/Tools/dotnetcli/dotnet.tar https://dotnetcli.blob.core.windows.net/dotnet/Sdk/1.0.0-preview3-003223/dotnet-dev-centos.7-x64.1.0.0-preview3-003223.tar.gz
17:26:49 tar: This does not look like a tar archive
17:26:49 tar: Exiting with failure status due to previous errors
```

File: init-tools.sh (hunk 1 above)
Comment (ianhays) [reply init-tools.sh]: Sounds good :)

**Comment by janvorli** [comment]
The tarballs for CentOS 7 and Ubuntu 14.04 are not on the URL. It seems that we will need the init-tools.sh from corefx after all - I think I've seen it using "Ubuntu" and "centos" without the version in the url. These exist.

**Comment by ianhays** [comment]
Yeah, I forgot about the package naming differences for centos, rhel, etc.

**Comment by ianhays** [comment]
Interesting! Updating the tools causes a very odd failure on Ubuntu1604:

```
17:58:27 CMake Error at src/ToolBox/SOS/lldbplugin/CMakeLists.txt:53 (message):
17:58:27   Cannot find lldb-3.5, lldb-3.6, lldb-3.8, lldb-3.9 or lldb-4.0.  Try
17:58:27   installing lldb-3.6-dev (or the appropriate package for your platform)
```

This failure is also happening in my attempted opensuse423 coreclr builds. @janvorli any ideas?

**Comment by weshaggard** [comment]
cc @jashook @RussKeldorph

**Comment by janvorli** [comment]
There is some problem with Windows build that causes the failure of all the other tests (except for Ubuntu 14.04 that fails due to missing lldb-3.6-dev package installation on the build machine)

**Comment by janvorli** [comment]
I've discovered the failure reason - with the new tools, the dactablegen.exe, InjectResource.exe and GenClrDebugResource.exe files are missing from the build tools dir. And the reason is that the init-tools.cmd that we've copied from corefx doesn't contain entry for restoring Microsoft.DotNet.BuildTools.Coreclr in PROJECT_JSON_CONTENTS.

**Comment by janvorli** [comment]
I have a fix for the Windows build issue. Since Ian is OOF, I'll create a new PR with the changes from this PR plus the fixes.

**Comment by janvorli** [comment]
I am closing the PR as I have created #17057  that supersedes it.

**Pull request closed without merging**